File: spring-boot-starter-data-requery/src/main/java/com/coupang/springframework/data/requery/starter/domain/model/AbstractPerson.java
package com.coupang.springframework.data.requery.starter.domain.model;

import io.requery.Entity;
import io.requery.Generated;
import io.requery.Key;
import io.requery.Transient;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;
import org.kotlinx.objectx.ToStringBuilder;
import org.springframework.data.requery.kotlin.domain.AbstractPersistable;
import org.springframework.data.requery.starter.dto.PersonDto;

import static org.kotlinx.core.HashxKt.hashOf;

/**
 * @author Diego on 2018. 7. 1..
 */

@Getter
@Entity
public class AbstractPerson extends AbstractPersistable<Long> {

    @Key
    @Generated
    protected Long id;

    protected String name;

    protected String email;

    @Override
    public int hashCode() {
        return hashOf(name, email);
    }

    @Transient
    @Override
    protected @NotNull ToStringBuilder buildStringHelper() {
        return super.buildStringHelper()
            .add("name", name)
            .add("email", email);
    }

    public PersonDto toPersonDto() {
        return PersonDto.of(name, email);
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractPerson.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.Column
import io.requery.Entity
import io.requery.Generated
import io.requery.Key

/**
 * AbstractPerson
 *
 * @author debop@coupang.com
 * @since 18. 5. 30
 */
@Entity(copyable = true, cacheable = true)
abstract class AbstractPerson: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    abstract val id: Long?

    @get:Column(name = "person_name", length = 48)
    abstract var name: String?

}